### ISSUE TITLE
Import: enable preview photo selection before import

### ISSUE THREAD

**Issue by coolstuff99** [photoprism/photoprism#943]
Hi,

I recently start using photoprism and it's exactly what I was looking for. Many thanks for continuous job done on this awesome tool.
I want to contribute with couple of ideas on the import side.

Scenario:
I have several phone devices with automatic photo import to an specific folder. The main problem is most of the time rubish pictures are automatically upload to the specific folder and I have to manually delete them and also do the same on my phones.

Problem:
I am using the import folder to upload all automatic photos but there is no way to remove/unselect those photos directly in photoprism to not be imported to "original" folder.

Proposal:
To include intermediate state for the user on the import section to be able to select what photos want to be imported.

Additional info:
In order to speed up the selection process all imported photos should be indexed before to process to select them.

Bonus:
The ones not selected because are rubish or simply doesn't want to be imported, a md5 could be created for each file to be marked as kind of "blocklist/denylist". In the future if same rubish appears again in import folder, this one automatically can be shown in grey in preview import selection process as "this was previously added to the blocklist/denylist".
I imagine this will require (md5/other hash method) files to be stored in database.
The grey photo behaviour can be used also to mark files as "already imported".

I can provide some mockups if the idea is interesting.

Thank you and congratulations again for the tool ;)

**Comment by tootbrute** [comment]
I'm not sure I agree with this proposal but I also have a similar problem.

**My workflow:**
1. Take pictures on my phone
2. AutoUploaded to my Nextcloud server
3. Delete the crap photos there and on my phone
4. Move the rest to Import folder on my server
5. Initiate the import

An easier thing for developers to implement might be a 'jail' that can be enabled for photos uploaded?
i.e. All photos must be approved before they are added to the main index. Then a user could check the metadata before letting those files join the rest of the photos.

I think once the "Delete" function is enabled on Photoprism, I won't care too much about step #3 in my process and will just automate everything.

**Comment by MafiaInc** [comment]
I don't thing the imort process should be complicated because there are billions of billions way to make your workflow better.
One  more idea, for how I do this. Sync my photos from phone to "originals" only when "syncthing" detects there is an album I have created in the phone. e. g. I just don't sync the default "Camera" directory. This how only "approved/selected" photos are going in. Albums that start with "my" are considered private for the phone and not uploaded.

Jan 23, 2021 08:10:23 tootbrute <notifications@github.com>:

> I'm not sure I agree with this proposal but I also have a similar problem.
> 
> My workflow:
> 
- * > Take pictures on my phone
- * > AutoUploaded to my Nextcloud server
- * > Delete the crap photos there and on my phone
- * > Move the rest to Import folder on my server
- * > Initiate the import

**Comment by tootbrute** [comment]
> I don't thing the imort process should be complicated because there are billions of billions way to make your workflow better. One  more idea, for how I do this. Sync my photos from phone to "originals" only when "syncthing" detects there is an album I have created in the phone. e. g. I just don't sync the default "Camera" directory. This how only "approved/selected" photos are going in. Albums that start with "my" are considered private for the phone and not uploaded. 

Thanks. Cool idea. I'll look at something like that for myself. I can save myself a step :)

**Comment by coolstuff99** [comment]
I agree my proposal can be refined but the general idea is:
- I have several devices with autoupload to make sure backup of all photos instantly. Therefore no preselection of any photo/video (to any album) because directly is uploaded
- Keep all uploaded content in a "import" or similar folder before migrate to original one.
- Review of the content is needed because of plenty of rubish, whatsapp images, etc...
- Enable some feature to detect if content in import has been already moved to original or marked previously as "non wanted content". This is why I proposed md5 functionality but can be any other method (don't know what's the trend now).

With this approach I don't care what's on my phone, everything will be uploaded as a backup and I can review it later. Also don't care if more than one device have same photo (because I shared using airdrop for instance) because will be reviewed and identified before moving to original.

**Comment by akdor1154** [comment]
I just set up as a new user and wanted something like this as well for the same reasons - 60% of my phone photos are junk.

The first thing I tried to check is if I can just send everything I import into the Review system, I thought that might also achieve something that would work for me. But it doesn't seem possible to set this up as far as I can see.

**Comment by graciousgrey** [comment]
@akdor1154 you are right, at the moment the review section only filters for low quality and non-photographic content. What you can do is sorting search results by recently added to see photos to be reviewed on top. 

In general I see two options:
1) We add an option that allows users to decide that all imported content needs to be approved in the review section. This way the imported content would be added to the index and unwanted content would be deleted  during the review.

2) We add an intermediate step. Newly imported content is shown in a review section but is not fully indexed. We probably do not need to run image classification, facial recognition, metadata extraction on this content before it get's approved. We also might not need to create all thumbnail sizes. 

What views and what information is needed for all of you to be able to review your files?

**Comment by akdor1154** [comment]
For me, no intermediate step would be fine. Would be useful if i could set "review all" at import time as well as a global option though.
